ionic2/app/components/poke-filter-pokemon-tab/poke-filter-pokemon-tab.component.ts

```diff
@@ -1,9 +1,117 @@
-import { Component } from '@angular/core';
+import { Component, OnInit, ChangeDetectionStrategy, Input, Output, EventEmitter } from '@angular/core';
+import { ApiService } from '../../services/api.service';
+import { Subscription } from 'rxjs';
+import { Pokemon } from '../../models/pokemon';
+import { PokemonFilterPipe } from './pokemon-filter-pipe';
+import { PokemonFilterData } from './pokemon-filter-data';
+import { Filter } from '../../models/filter';
+
+type PokemonContainer = {pokemon: Pokemon, isSelected: boolean};
+type TypeContainer = {type: string, isSelected: boolean};
 
 @Component({
+  template: require('./poke-filter-pokemon-tab.component.html'),
   selector: 'poke-filter-pokemon-tab',
-  template: `<ion-content>Pokemon</ion-content>`
+  pipes: [ PokemonFilterPipe ],
+  changeDetection: ChangeDetectionStrategy.OnPush
```

Comment:
When I built the pipe i tested it and it didnt work - after reading about this ChangeDetectionStrategy I tried it and it worked. I had the problem that the pipe is not executed again if I change the parameters of the pipe.

Now after trying again without that it still works ... so I guess it can be removed.